### PR TITLE
[mercadopago-sdk-js] Improve params and returns from constructors and submits

### PR DESCRIPTION
**Pull request by francineemilia**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:
If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
For changes in constructors, they were made based on this doc: https://www.mercadopago.com/developers/en/docs/checkout-bricks/common-initialization
For changes related to submits, they were made based on these docs:
https://www.mercadopago.com/developers/en/docs/checkout-bricks/status-screen-brick/default-rendering
https://www.mercadopago.com/developers/en/docs/checkout-bricks/wallet-brick/default-rendering
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Comment by AleFossati** [comment]
Fran, acho que o `format` deu uma dificultada novamente para identificarmos o que é alteração e o que é formatação. Poderia tirar os ajustes de formatação, pls?

Eu desativaria o `format on save` do vscode e faria o reset da branch pra três commits atrás, descartando os três do print. Assim daria pra fazê-los novamente sem formatação.


image

**Review by meliguicarvalho — APPROVED** [review]

File: types/mercadopago-sdk-js/modules/bricks.d.ts [hunk 1]
```diff
@@ -1,328 +1,335 @@
...
-      settings: BrickSettings
-    ): Promise<BrickController>;
-  }
+    interface BrickError {
+        type: 'non_critical' | 'critical';
+        cause: string;
+        message: string;
+    }
+
+    interface iBrickCallbacks {
```
Comment (AleFossati) [inline types/mercadopago-sdk-js/modules/bricks.d.ts]:
```suggestion
    interface BrickCallbacks {
```

File: types/mercadopago-sdk-js/modules/bricks.d.ts [hunk 2]
```diff
@@ -1,328 +1,335 @@
...
+        onReady?: () => void;
+        onError?: (error: BrickError) => void;
+    }
+
+    interface onSubmit<BrickType> {
```
Comment (AleFossati) [inline types/mercadopago-sdk-js/modules/bricks.d.ts]: podemos remover o `on` da interface

```suggestion
    interface Submit<BrickType> {
```

File: types/mercadopago-sdk-js/modules/bricks.d.ts [hunk 3]
```diff
@@ -1,328 +1,335 @@
...
+        onSubmit: (
+            formData: CardFormData | PaymentFormData,
+            param2?: AdditionalCardFormData | AdditionalPaymentFormData,
+        ) => BrickType extends 'wallet' ? Promise<string> : Promise<void>;
+    }
+
+    interface onBinChange {
```
Comment (AleFossati) [inline types/mercadopago-sdk-js/modules/bricks.d.ts]: podemos remover o `on` da interface

```suggestion
    interface BinChange {
```

**Comment by AleFossati** [comment]
Show Fran, ficou bem mais fácil, obrigado!
Só tenho algumas sugestões de nomenclatura pra gente tentar manter um padrão, geralmente tipos e interfaces a gente usa `PascalCase` e conversamos com core-cho recentemente pra não adicionarmos `I` e `T` antes dos nomes dos tipos e interfaces.

File: types/mercadopago-sdk-js/modules/bricks.d.ts [hunk 4]
```diff
@@ -1,328 +1,335 @@
...
+
+    interface BrickCallbacks {
+        onReady?: () => void;
+        onError?: (error: BrickError) => void;
+    }
+
+    interface Submit<BrickType> {
+        onSubmit: (
+            formData: CardFormData | PaymentFormData,
+            param2?: AdditionalCardFormData | AdditionalPaymentFormData,
```
Comment (ghost) [inline types/mercadopago-sdk-js/modules/bricks.d.ts]: Na documentação nós chamamos esse parâmetro de `additionalData`
```suggestion
            additionalData?: AdditionalCardFormData | AdditionalPaymentFormData,
```

File: types/mercadopago-sdk-js/modules/bricks.d.ts [hunk 5]
```diff
@@ -1,328 +1,335 @@
...
+        PAY = 'pay',
+        BUY = 'buy',
+    }
+
+    enum WalletButtonValueProp {
+        PRACTICALITY = 'practicality',
+        CONVENIENCE = 'convenience',
+        SECURITY_DETAILS = 'security_details',
+        SECURITY_SAFETY = 'security_safety',
+    }
```
Comment (ghost) [inline types/mercadopago-sdk-js/modules/bricks.d.ts:100]: Aqui estão faltando as novas value props, não?  `convenience_all`, `convenience_credits` e `smart_option`.
cc @meliguicarvalho para verificar.

Comment (francineemilia) [reply types/mercadopago-sdk-js/modules/bricks.d.ts:100]: @andreagostinho-meli no caso das novas value props, elas estão no commit do Isma, esta PR é anterior ao release, daí a diferença de escopo do PR

**Review by AleFossati — APPROVED** [review]

**Pull request merged**